plugins/govppmux/govpp_channel.go

```diff
@@ -27,16 +38,56 @@ type govppRequestCtx struct {
 	requestMsg govppapi.Message
 	// Retry data
 	retry retryConfig
+	// Stopwatch object
+	stopwatch *measure.Stopwatch
+	// Current duration
+	started time.Time
 }
 
-// helper struct holding info about retry configuration
-type retryConfig struct {
-	attempts int
-	timeout  time.Duration
+// govppMultirequestCtx is custom govpp MultiRequestCtx.
+type govppMultirequestCtx struct {
+	// Original multi request context
+	requestCtx govppapi.MultiRequestCtx
+	// Function allowing to re-send request in case it's granted by the config file
+	sendRequest func(govppapi.Message) govppapi.MultiRequestCtx
+	// Parameter for sendRequest
+	requestMsg govppapi.Message
+	// Stopwatch object
+	stopwatch *measure.Stopwatch
+	// Current duration
+	started time.Time
+}
+
+// SendRequest sends asynchronous request to the vpp and receives context used to receive reply.
+// Plugin govppmux allows to re-send retry which failed because of disconnected vpp, if enabled.
+func (c *goVppChan) SendRequest(request govppapi.Message) govppapi.RequestCtx {
+	startTime := time.Now()
+
+	logrus.DefaultLogger().Warnf("request sent %v", request.GetMessageName())
+	sendRequest := c.Channel.SendRequest
+	// Send request now and wait for context
+	requestCtx := sendRequest(request)
+
+	// Return context with value and function which allows to send request again if needed
+	return &govppRequestCtx{
+		requestCtx:  requestCtx,
+		sendRequest: sendRequest,
+		requestMsg:  request,
+		retry:       c.retry,
+		stopwatch:   c.stopwatch,
+		started:     startTime,
+	}
 }
 
 // ReceiveReply handles request and returns error if occurred. Also does retry if this option is available.
 func (r *govppRequestCtx) ReceiveReply(reply govppapi.Message) error {
+	defer func(t time.Time) {
+		if r.stopwatch != nil {
+			r.stopwatch.TimeLog(r.requestMsg.GetMessageName()).LogTimeEntry(time.Since(r.started))
+			r.stopwatch.PrintLog()
```

Comment:
Why are we printing here?

Reply:
stored times are not printed anywhere now, only in ReceiveReply() for both, single and multi-request

the alternative is to print the log after every resync/crud, what do you think?

Reply:
It will kind of _spammy_ to print times for each request. I think it would be best to make the times available through REST API for retrieval anytime.